pkg/controllers/uiplugin/monitoring.go

```diff
@@ -2,39 +2,79 @@ package uiplugin
 
 import (
 	"fmt"
+	"slices"
 	"strings"
 
 	osv1 "github.com/openshift/api/console/v1"
 	osv1alpha1 "github.com/openshift/api/console/v1alpha1"
+	"golang.org/x/mod/semver"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 
 	uiv1alpha1 "github.com/rhobs/observability-operator/pkg/apis/uiplugin/v1alpha1"
 )
 
-func createMonitoringPluginInfo(plugin *uiv1alpha1.UIPlugin, namespace, name, image string, features []string) (*UIPluginInfo, error) {
-	config := plugin.Spec.Monitoring
-	if config == nil {
-		return nil, fmt.Errorf("monitoring configuration can not be empty for plugin type %s", plugin.Spec.Type)
+var AcmErrorMsg = `if you intend to enable "acm-alerting" thanos querier url and alertmanager url needs to be configured in the custom resource UIPlugin. `
+var PersesErrorMsg = `if you intend to enable "perses-dashboards" a perses service name and namespace needs to be configured in the custom resource UIPlugin.`
+var AlertmanagerEmptyMsg = "alertmanager location is empty for plugin type monitoring."
+var ThanosEmptyMsg = "thanosQuerier location is empty for plugin type monitoring."
+var PersesNameEmptyMsg = "persesName location is empty for plugin type monitoring."
+var PersesNamespaceEmptyMsg = "persesNamespace location is empty for plugin type monitoring."
+var IncompatibleFeaturesAndConfigsErrorMsg = "UIPlugin configurations are incompatible with feature flags"
+
+func getConfigError(config *uiv1alpha1.MonitoringConfig) (bool, bool, bool, string) {
```

Comment:
this function is probably doing too much, maybe it makes sense to split this per main feature (acm and perses) and based on the results add the respective feature

Reply:
Separation of concerns. Split configuration validation for acm, perses, and incidents into their own functions.